File: cpp/leetcode/leetcode/28_implement_strstr.cpp
//
//  28_implement_strstr.cpp
//  leetcode
//
//  Created by hatlonely on 16/1/13.
//  Copyright © 2016年 hatlonely. All rights reserved.
//
//  Implement strStr().
//
//  Returns the index of the first occurrence of needle in haystack,
//  or -1 if needle is not part of haystack.
//

#include <iostream>
#include <cassert>
#include <string>

namespace implement_strstr {
    
class Solution {
public:
    int strStr(std::string source, std::string pattern) {
        size_t pos = source.find(pattern);
        if (pos == std::string::npos) {
            return -1;
        }
        return (int)pos;
    }
};

int main(int argc, char *argv[]) {
    return 0;
}
    
}

File: go/src/main/28_implement_strstr.go
package main

import (
	"fmt"
	"strings"
)

func strStr(source string, pattern string) int {
	return strings.Index(source, pattern)
}

func main() {
	for _, unit := range []struct {
		source  string
		pattern string
		index   int
	}{
		{"1234", "23", 1},
	} {
		if index := strStr(unit.source, unit.pattern); index != unit.index {
			fmt.Printf("strStr(%v, %v) = %v != %v\n", unit.source, unit.pattern, index, unit.index)
		}
	}
}
